sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py

```diff
@@ -1087,43 +1502,333 @@ def test_create_thread_and_run(self, **kwargs):
+            assert run.id
+            print("Created run, run ID", run.id)
+
+            if run.status == "failed":
+                assert run.last_error
+                print(run.last_error)
+                print("FAILED HERE")
+
+            # check status
+            assert run.status in ["queued", "in_progress", "requires_action", "completed"]
+            while run.status in ["queued", "in_progress", "requires_action"]:
+                # wait for a second
+                time.sleep(1)
+                run = client.agents.get_run(thread_id=thread.id, run_id=run.id)
+                if run.status == "failed":
+                    assert run.last_error
+                    print(run.last_error)
+                    print("FAILED HERE")
+                assert run.status in [
+                    "queued",
+                    "in_progress",
+                    "requires_action",
+                    "completed",
+                ]
+                print("Run status:", run.status)
+
+            # list steps, check that get_run_step works with first step_id
+            steps = client.agents.list_run_steps(thread_id=thread.id, run_id=run.id)
+            assert steps["data"].__len__() > 0
+            step = steps["data"][0]
+            get_step = client.agents.get_run_step(
+                thread_id=thread.id, run_id=run.id, step_id=step.id
+            )
+            assert step == get_step
+
+            # delete agent and close client
+            client.agents.delete_agent(agent.id)
+            print("Deleted agent")
+        
+    
+    
+    
+    # # **********************************************************************************
+    # #
+    # #                      HAPPY PATH SERVICE TESTS - Streaming APIs
+    # #
+    # # **********************************************************************************
+
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_create_stream(self, **kwargs):
+        """Test creating stream."""
+
+        # create client
+        with self.create_client(**kwargs) as client:
+            assert isinstance(client, AIProjectClient)
+
+            # create agent
+            agent = client.agents.create_agent(model="gpt-4o", name="my-agent", instructions="You are helpful agent")
+            assert agent.id
+            print("Created agent, agent ID", agent.id)
+
+            # create thread
+            thread = client.agents.create_thread()
+            assert thread.id
+            print("Created thread, thread ID", thread.id)
+
+            # create message
+            message = client.agents.create_message(thread_id=thread.id, role="user", content="Hello, can you tell me a joke?")
+            assert message.id
+            print("Created message, message ID", message.id)
+
+            # create stream
+            with client.agents.create_stream(thread_id=thread.id, assistant_id=agent.id) as stream:
+                assert isinstance(stream, AgentRunStream)
```

Comment:
This assert should fail.    The new design return AgentEventHandler or AsyncAgentEventHandler.  But it looks liked the pipleine pass.   So I am checking why it pass also.